### PR TITLE
hydron: unstable-2019-02-17 -> 3.0.4

### PR DESCRIPTION
**Pull request by freezeboy**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Chiiruno** [comment]
LGTM, but not on NixOS anymore so can't test.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

File: pkgs/servers/hydron/default.nix [hunk 1]
```diff
@@ -1,17 +1,18 @@
 { stdenv, buildGoPackage, fetchFromGitHub, pkgconfig, ffmpeg-full, graphicsmagick
 , quicktemplate, go-bindata, easyjson }:
 
-buildGoPackage {
-  pname = "hydron-unstable";
-  version = "2019-02-17";
+buildGoPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/servers/hydron/default.nix:4]: Can we use buildGoModule?

**Comment by Chiiruno** [comment]
That reminds me, you may want to reference this PR.
[#67666](https://github.com/NixOS/nixpkgs/pull/67666)

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107810` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hydron</li>
  </ul>
</details>

**Comment by freezeboy** [comment]
Trying to use buildGoModule right now

**Comment by freezeboy** [comment]
quicktemplate and statik looks ok, but hydron has problems

hydron case:
# github.com/chai2010/webp
vendor/github.com/chai2010/webp/capi.go:24:10: fatal error: webp.h: No such file or directory
   24 | #include "webp.h"
      |          ^~~~~~~~
compilation terminated.

and I am not sure why this happens with modules and not packages

File: pkgs/servers/hydron/default.nix [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
   src = fetchFromGitHub {
     owner = "bakape";
     repo = "hydron";
-    rev = "824789fb108966432e507143db39b358dd7ff233";
-    sha256 = "1xxykjf5iyavm12gd6nx4j8x2mlzzn7x8vm0j5009lsir98qr5zn";
+    rev = "v${version}";
+    sha256 = "BfMkKwz7ITEnAIMGUHlBH/Dn9yLjWKoqFWupPo1s2cs=";
   };
 
   nativeBuildInputs = [ pkgconfig go-bindata ];
```
Comment (SuperSandro2000) [inline pkgs/servers/hydron/default.nix:18]:
```suggestion
  nativeBuildInputs = [ pkg-config go-bindata ];
```

File: pkgs/servers/hydron/default.nix [hunk 3]
```diff
@@ -1,17 +1,18 @@
 { stdenv, buildGoPackage, fetchFromGitHub, pkgconfig, ffmpeg-full, graphicsmagick
```
Comment (SuperSandro2000) [inline pkgs/servers/hydron/default.nix:1]:
```suggestion
{ stdenv, buildGoPackage, fetchFromGitHub, pkg-config, ffmpeg-full, graphicsmagick
```

**Comment by SuperSandro2000** [comment]
> quicktemplate and statik looks ok, but hydron has problems
> 
> hydron case:
> 
> # github.com/chai2010/webp
> vendor/github.com/chai2010/webp/capi.go:24:10: fatal error: webp.h: No such file or directory
> 24 | #include "webp.h"
> | ^~~~~~~~
> compilation terminated.
> 
> and I am not sure why this happens with modules and not packages

I think buildGoModule could properly isolate the build dependencies.

**Comment by SuperSandro2000** [comment]
Not sure why the header file (https://github.com/chai2010/webp/blob/master/internal/include/webp.h) is deleting while vendoring.

**Comment by SuperSandro2000** [comment]
```
diff --git a/pkgs/servers/hydron/default.nix b/pkgs/servers/hydron/default.nix
index 18dba9ce55e..1fd56e6fd2d 100644
--- a/pkgs/servers/hydron/default.nix
+++ b/pkgs/servers/hydron/default.nix
@@ -1,13 +1,11 @@
-{ stdenv, buildGoPackage, fetchFromGitHub, pkgconfig, ffmpeg-full, graphicsmagick
-, quicktemplate, go-bindata, easyjson }:
+{ stdenv, buildGoModule, fetchFromGitHub
+, pkg-config, ffmpeg
+, graphicsmagick, quicktemplate, go-bindata, easyjson }:

-buildGoPackage rec {
+buildGoModule rec {
   pname = "hydron";
   version = "3.0.4";

-  goPackagePath = "github.com/bakape/hydron";
-  goDeps = ./deps.nix;
-
   src = fetchFromGitHub {
     owner = "bakape";
     repo = "hydron";
@@ -15,9 +13,13 @@ buildGoPackage rec {
     sha256 = "BfMkKwz7ITEnAIMGUHlBH/Dn9yLjWKoqFWupPo1s2cs=";
   };

-  nativeBuildInputs = [ pkgconfig go-bindata ];
+  vendorSha256 = "1ngig5zw0gf1mkjjsfvvn09rncb36rg274cbi3glp8wzfcr8aip3";

-  buildInputs = [ ffmpeg-full graphicsmagick quicktemplate easyjson ];
+  runVend = true;
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ ffmpeg ];

   meta = with stdenv.lib; {
     homepage = "https://github.com/bakape/hydron";
```

**Comment by freezeboy** [comment]
Thank you I applied it, (and removed some unused deps)

**Comment by SuperSandro2000** [comment]
@freezeboy please fix the merge conflict.

File: pkgs/servers/hydron/default.nix [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
-  nativeBuildInputs = [ pkgconfig go-bindata ];
+  nativeBuildInputs = [ pkg-config ];
 
-  buildInputs = [ ffmpeg-full graphicsmagick quicktemplate easyjson ];
+  vendorSha256 = "1ngig5zw0gf1mkjjsfvvn09rncb36rg274cbi3glp8wzfcr8aip3";
+  runVend = true;
+
+  buildInputs = [ ffmpeg ];
 
   meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/servers/hydron/default.nix:21]:
```suggestion
  meta = with lib; {
```

Comment (freezeboy) [reply pkgs/servers/hydron/default.nix]: done

File: pkgs/servers/hydron/default.nix [hunk 5]
```diff
@@ -1,22 +1,24 @@
 { lib, stdenv, buildGoPackage, fetchFromGitHub, pkgconfig, ffmpeg-full, graphicsmagick
 , quicktemplate, go-bindata, easyjson }:
+{ lib, stdenv, buildGoModule, fetchFromGitHub, pkg-config, ffmpeg, graphicsmagick }:
```
Comment (SuperSandro2000) [inline pkgs/servers/hydron/default.nix:3]:
```suggestion
{ lib, stdenv, buildGoModule, fetchFromGitHub, pkg-config, ffmpeg }:
```

graphicsmagick is no longer used.

**Comment by freezeboy** [comment]
Fixed

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107810` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>hydron</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107810` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>hydron</li>
</ul>
</details>

File: pkgs/servers/hydron/deps.nix [hunk 6]
```diff
@@ -1,65 +1,92 @@
-# This file was generated by https://github.com/kamilchm/go2nix v1.3.0
```
Comment (SuperSandro2000) [inline pkgs/servers/hydron/deps.nix:1]: Please remove this file as buildGoModule does not need it.

**Review by SuperSandro2000 — APPROVED** [review]
other than that LGTM

**Pull request merged**